### ISSUE TITLE
Change request and response field snippets so that descendants are not documented by default

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-restdocs#274]
See http://stackoverflow.com/questions/38240818/how-do-i-make-sure-my-documentation-is-up-to-date-with-spring-rest-docs for details. The summary is that a field documenting its descendants makes it easy to miss a newly added descendant field.

**Comment by andreasevers** [comment]
To pick up the discussion from SO:

```json
{
"car": {
    "motor" : {
        "brand": "Porsche",
        "power": "165 kW"
    },

    "suspension": {
        "type": "automatic"
    }
}
```

Perhaps the best solution here would be that all siblings of a documented field should be documented?
So in that case if you document car.motor, car.suspension should be documented as well - but not any subfield of these two. If you document car.motor.brand, then you have to document car.motor.power as well. Consequently, if a new field is added in the payload, it will always be breaking your test unless you never documented a single field on the level it is added to.

**Comment by otrosien** [comment]
Alternative proposal:

If you want to document an element including sub-elements, you make it explicit. Given the example above either `fieldWithPath("car.*").description("a car")` or `fieldWithPath("car").description("a car")`, whereas in the second case you'll still need to document the sub-keys. But, this is definitely a breaking change to the existing behaviour.

**Comment by wilkinsona** [comment]
My current thinking is that the default behaviour should be changed and a new method should be added to `FieldDescriptor` to explicitly state that it documents its descendants. This will be a breaking change, although arguably one that fixes a bug, so it'll have to wait for 1.2 with appropriate migration instructions for those who were intentionally making use of the current behaviour.

**Comment by andreasevers** [comment]
That sounds useful, clear and fair.
